### ISSUE TITLE
Cannot compile sysdig/libscap bpf driver in 6.6.4-zen kernel

### ISSUE THREAD

**Issue by unknowndevQwQ** [falcosecurity/libs#1562]
**Describe the bug**
When trying to use `sysdig --bpf ... ` I was unable to download the prebuilt bpf probe, and compiling the bpf probe resulted in the error.

**How to reproduce it**
run `sysdig --bpf ...`

**Expected behaviour**
*Capturing events with sysdig.

**Screenshots**

```
* Running scap-driver-loader for: driver version=0.13.1, arch=x86_64, kernel release=6.6.4-zen1-1-zen, kernel version=1
* Running scap-driver-loader with: driver=bpf, compile=yes, download=yes
* Filename 'scap_arch_6.6.4-zen1-1-zen_1.o' is composed of:
 - driver name: scap
 - target identifier: arch
 - kernel release: 6.6.4-zen1-1-zen
 - kernel version: 1
* Trying to download a prebuilt eBPF probe from https://download.sysdig.com/scap-drivers/0.13.1/x86_64/scap_arch_6.6.4-zen1-1-zen_1.o
curl: (22) The requested URL returned error: 404
Unable to find a prebuilt scap eBPF probe
* Trying to compile the eBPF probe (scap_arch_6.6.4-zen1-1-zen_1.o)
In file included from /usr/src/scap-0.13.1/bpf/probe.c:16:
In file included from ./include/linux/sched.h:12:
In file included from ./arch/x86/include/asm/current.h:9:
In file included from ./include/linux/cache.h:6:
In file included from ./arch/x86/include/asm/cache.h:5:
In file included from ./include/linux/linkage.h:8:
In file included from ./arch/x86/include/asm/linkage.h:6:
./arch/x86/include/asm/ibt.h:77:8: warning: 'nocf_check' attribute ignored; use -fcf-protection to enable the attribute [-Wignored-attributes]
extern __noendbr u64 ibt_save(bool disable);
       ^
./arch/x86/include/asm/ibt.h:32:34: note: expanded from macro '__noendbr'
#define __noendbr       __attribute__((nocf_check))
                                       ^
./arch/x86/include/asm/ibt.h:78:8: warning: 'nocf_check' attribute ignored; use -fcf-protection to enable the attribute [-Wignored-attributes]
extern __noendbr void ibt_restore(u64 save);
       ^
./arch/x86/include/asm/ibt.h:32:34: note: expanded from macro '__noendbr'
#define __noendbr       __attribute__((nocf_check))
                                       ^
In file included from /usr/src/scap-0.13.1/bpf/probe.c:26:
/usr/src/scap-0.13.1/bpf/fillers.h:2844:33: error: use of undeclared identifier 'I_CTIME_QUERIED'
        time.tv_nsec = time.tv_nsec & ~I_CTIME_QUERIED; // See https://elixir.bootlin.com/linux/v6.6-rc1/source/include/linux/fs.h#L1544
                                       ^
2 warnings and 1 error generated.
make[3]: *** [/usr/src/scap-0.13.1/bpf/Makefile:54: /usr/src/scap-0.13.1/bpf/probe.o] Error 1
make[2]: *** [/usr/lib/modules/6.6.4-zen1-1-zen/build/Makefile:1911: /usr/src/scap-0.13.1/bpf] Error 2
make[1]: *** [Makefile:234: __sub-make] Error 2
make: *** [Makefile:38: all] Error 2
mv: cannot stat '/usr/src/scap-0.13.1/bpf/probe.o': No such file or directory
Unable to load the scap eBPF probe
Unable to load the BPF probe
can't open BPF probe '/root/.scap/scap-bpf.o'
```

**Environment**

- Falco version: not installed falco, from sysdig: `sysdig version 0.34.1`

- System info: not installed falco, none

- Cloud provider or hardware configuration: VirtualBox 7.0.12
- OS: `Arch`

- Kernel: `Linux arch 6.6.4-zen1-1-zen #[null]1 ZEN SMP PREEMPT_DYNAMIC Mon, 04 Dec 2023 00:28:58 +0000 x86_64 GNU/Linux`

- Installation method: `pacman`

**Additional context**
If I try to run `sysdig proc.name in (podman, iptables) and evt.type in (execve)` using the dkms module, then I'll experience lockups starting with `[rcub/0]` and `[kworker/2:3+rcu_gp]` after capturing the event, and then systemd as well!

**Comment by FedeDP** [comment]
Hi! This is a known issue since driver 6.0.1+driver (the latest tagged one) contained a fix for the build of Linux 6.6.0-rc{1-3}.
Unfortunately , the patch was then reverted during the Linux 6.6 release cycle (at the rc4 stage) but then we had already released driver 6.0.1+driver.
Therefore, libs master compiles great against Linux 6.6, but latest tag does not.
Basically we were too quick to react to Linux changes this time...

**Comment by FedeDP** [comment]
Also, later the patches were re-applied in linux 6.7 and thus currently the build of drivers is broken against it: https://github.com/falcosecurity/libs/actions/workflows/latest-kernel.yml.

Original patch that got merged in our drivers to fix the build: https://github.com/falcosecurity/libs/pull/1349
Partial revert of the aforementioned patch to re-fix the build against linux 6.6.0-rc3+: https://github.com/falcosecurity/libs/pull/1415

**Comment by unknowndevQwQ** [comment]
> Also, later the patches were re-applied in linux 6.7 and thus currently the build of drivers is broken against it: https://github.com/falcosecurity/libs/actions/workflows/latest-kernel.yml. This time, we learnt the lesson and are waiting a little bit more in the linux 6.7 development stage before applying build fixes (at least rc5 let's say).
> 
> Original patch that got merged in our drivers to fix the build: #1349 Partial revert of the aforementioned patch to re-fix the build against linux 6.6.0-rc3+: #1415

What should I do if I want to get sysdig working again? As well, is the kernel deadlock problem a known issue? If not, should I open a new issue?

**Comment by FedeDP** [comment]
> What should I do if I want to get sysdig working again?

You should wait for the next release or use a development sysdig version :/ not much we can do about those kind of things (ie: it's pretty hard to support future kernel versions at the moment we release; i tried to apply the fix for linux 6.6 before the `6.0.1+driver` tag asap for this exact reason, ie: to tag a driver version that would work with a kernel release that was ~1months away).

> As well, is the kernel deadlock problem a known issue? If not, should I open a new issue?

nope, we should never deadlock! Please open a new issue, thank you very much!

**Comment by FedeDP** [comment]
I will close this one since this is expected issue :) 
/close
